documentation/homework/week1/todo-app/src/components/Todo-item.js

```diff
@@ -0,0 +1,42 @@
+import React, { Component } from 'react'
+
+export default class TodoItem extends Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      checked: this.props.done,
+      styleCheck: {
+        textDecoration: null,
+        color: null
+      }
+    }
+    this.handleCheck = this.handleCheck.bind(this)
+  }
+  handleCheck(e) {
+    const checked = e.target.checked
+    this.setState(prevState => ({
+      checked: !prevState.checked,
+      styleCheck: {
+        textDecoration: checked && 'line-through',
+        color: checked && 'red'
```

Comment:
Think carefully about this. Why do you need these `styleCheck` properties in your state? Both of them simply respond to the `checked` property, which is already in your state... You are setting two other properties based on `checked` for no reason.

You can update your styling in response to state changes inside the `render` function:

    <li style={this.state.checked ? something : somethingElse}>

Also as a general rule, it's a very bad idea to have CSS declarations inside component state, those two things should not be mixed.